doc/Data Plane Code Generation/Sample.md

```diff
@@ -0,0 +1,28 @@
+Samples are where developers could start to learn a library. You should generate samples scaffold by running [quick start tool](AzureSDKCodeGeneration_DataPlane_Quickstart.md#create-starter-package). The samples should match the structure containing the following key features:
+```
+├── ...
+├── samples
+│   ├── README.md
+│   ├── Sample1_{Scenario1}.md
+│   |── Sample1_{Scenario1}Async.md
+│   ├── Sample2_{Scenario2}.md
+│   |── Sample2_{Scenario2}Async.md
+│   └── ...
+├── tests
+│   ├── Samples
+|   |   ├── Sample1_{Scenario1}.cs
+|   |   ├── Sample1_{Scenario1}Async.cs
+|   |   ├── Sample2_{Scenario2}.cs
+|   |   ├── Sample2_{Scenario2}Async.cs
+|   |   └── ...
+│   └── ...
+└── ...
```

Comment:
This is a way, certainly, but more recently I've been putting sync and async methods in the same source file as well as showing examples of either in the markdown sample files. Since each sample markdown file typically has some introductory text and resource provisioning, all that would otherwise need to be duplicated.

As such, I recommend what sdk/template/Azure.Template shows now: both sync and async methods in a single file. See sdk/template/Azure.Template/samples/Sample1_HelloWorld.md and sdk/template/Azure.Template/test/samples/MiniSecretClientSamples.HelloWorld.cs (though, your suggestion of calling this Sample1_*.cs is better).

Reply:
I do agree with you. But the struct follows guidelines in https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-samples
[✅](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-sync-samples-naming) DO use synchronous APIs in the Sample1_HelloWorld.md sample. Add a second sample named Sample1_HelloWorldAsync.md that does the same thing as Sample1_HelloWorld.md using asynchronous code.

Reply:
You could, but I suggest using `partial` classes then so you don't have to keep duplicating class declaration e.g.

```csharp
// Sample1_HelloWorld.cs
public partial class Sample1 : SampleBase<SampleTestEnvironent>
{
  [Test]
  [SyncOnly]
  public void DoWork()
  {
    #region Azure_Sample_Sample1_DoWork
    // ...
    #endregion
}

// Sample1_HelloWorldAsync.cs
public partial class Sample1
{
  [Test]
  [AsyncOnly]
  public async Task DoWorkAsync()
  {
    #region Azure_Sample_Sample1_DoWorkAsync
    // ...
    #endregion
  }
}
```

That said, I can change the guidelines to match. Creating lots of extra files seems to offer no benefit.

Reply:
Sorry, I didn't get. The example you provided above still needs two files. 
I'm OK to have only one file for both sync and async methods as we already did in sdk/template/Azure.Template, if you could change the guidelines.

Reply:
I'm saying, if you want two files, you could use partial classes that way. Whether one file or two, though, they should derive from `SampleBase<T>` and have distinct sections for sync and async examples.